app/src/main/java/com/jxw/git_hub_users/view/DetailViewActivity.java

```diff
@@ -17,5 +29,70 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
+
+        if (savedInstanceState != null) {
+            this.ghUserInfo = savedInstanceState.getParcelable(USER_INFO_KEY);
+            this.displayUserProfile(this.ghUserInfo);
+        } else {
+            // HANDLING INCOMING INTENT
+            Intent incomingIntent = getIntent();
+            if (incomingIntent.hasExtra("userName")) {
+                String userName = incomingIntent.getStringExtra("userName");
+
+                UserProfilePresenter profilePresenter = new UserProfilePresenter(this);
+                profilePresenter.getUserProfile(userName);
+            } else {
+                Toast toast = Toast.makeText(getApplicationContext(), "Invaid Username ID, Please try again", Toast.LENGTH_LONG);
+                toast.show();
+            }
+        }
+
+        Log.i(TAG, "ON-CREATE IS CALLED");
+
+
+    }
+
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+
+        outState.putParcelable(USER_INFO_KEY, this.ghUserInfo);
+        Log.i(TAG, "ON-SAVE-INSTANCE-STATE IS CALLED");
+    }
+
+    @Override
+    public void displayUserProfile(UserProfile userProfile) {
```

Comment:
@johngorithm Kindly refactor this method to have at most 25 lines of code.

Reply:
@petermwash 
I have implemented this change in the PR that follows this one since I `cherry-picked` this into that one.
[PR with this fix](https://github.com/johngorithm/github-ap-with-java/pull/15)

Thanks